fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt

```diff
@@ -285,17 +319,24 @@ class DefaultTabsTrayController(
             }
             TabsTray.closedExistingTab.record(TabsTray.ClosedExistingTabExtra(source ?: "unknown"))
         }
+
+        tabsTrayStore.dispatch(TabsTrayAction.ExitSelectMode)
     }
 
-    /**
-     * Deletes a list of [tabs] offering an undo option.
-     *
-     * @param tabs List of [TabSessionState]s (sessions) to be removed.
-     * This method has no effect for tabs that do not exist.
-     */
-    override fun handleMultipleTabsDeletion(tabs: Collection<TabSessionState>) {
+    override fun handleDeleteSelectedTabsClicked() {
+        val tabs = tabsTrayStore.state.mode.selectedTabs
+
         TabsTray.closeSelectedTabs.record(TabsTray.CloseSelectedTabsExtra(tabCount = tabs.size))
 
+        deleteMultipleTabs(tabs)
+
+        tabsTrayStore.dispatch(TabsTrayAction.ExitSelectMode)
+    }
+
+    /**
+     * Helper function to delete multiple tabs and offer an undo option.
+     */
+    internal fun deleteMultipleTabs(tabs: Collection<TabSessionState>) {
```

Comment:
So this was tricky, and I knew I'd get a comment about it 😅. The issue was the original function, `handleMultipleTabsDeletion` was ONLY ever used to delete selected tabs, so rather than try to rework a bunch of tests & logic, I opted to refactor that to a helper function, `deleteMultipleTabs`, and then create `handleDeleteSelectedTabsClicked` to call that and handle the selected tabs-specific functionality (i.e. report metrics and grab the group of selected tabs.

Reply:
I added the `VisibleForTesting` annotation to this function. It by default has `otherwise = PRIVATE`